src/PerformanceSensitiveAnalyzers/CSharp/CallSiteImplicitAllocationAnalyzer.cs

```diff
@@ -42,65 +40,33 @@ internal sealed class CallSiteImplicitAllocationAnalyzer : AbstractAllocationAna
 
-        protected override void AnalyzeNode(SyntaxNodeAnalysisContext context, in PerformanceSensitiveInfo info)
+        protected override void AnalyzeNode(OperationAnalysisContext context, in PerformanceSensitiveInfo info)
         {
-            var node = context.Node;
-            var semanticModel = context.SemanticModel;
-            Action<Diagnostic> reportDiagnostic = context.ReportDiagnostic;
-            var cancellationToken = context.CancellationToken;
-
-            var invocationExpression = node as InvocationExpressionSyntax;
-
-            if (semanticModel.GetSymbolInfo(invocationExpression, cancellationToken).Symbol is IMethodSymbol methodInfo)
+            if (!(context.Operation is IInvocationOperation invocation))
             {
-                if (methodInfo.IsOverride)
-                {
-                    CheckNonOverridenMethodOnStruct(methodInfo, reportDiagnostic, invocationExpression);
-                }
-
-                if (methodInfo.Parameters.Length > 0 && invocationExpression.ArgumentList != null)
-                {
-                    var lastParam = methodInfo.Parameters[methodInfo.Parameters.Length - 1];
-                    if (lastParam.IsParams)
-                    {
-                        CheckParam(invocationExpression, methodInfo, semanticModel, reportDiagnostic, cancellationToken);
-                    }
-                }
+                return;
             }
-        }
 
-        [MethodImpl(MethodImplOptions.NoInlining)]
-        private static void CheckParam(InvocationExpressionSyntax invocationExpression, IMethodSymbol methodInfo, SemanticModel semanticModel, Action<Diagnostic> reportDiagnostic, CancellationToken cancellationToken)
-        {
-            var arguments = invocationExpression.ArgumentList.Arguments;
-            if (arguments.Count != methodInfo.Parameters.Length)
-            {
-                reportDiagnostic(Diagnostic.Create(ParamsParameterRule, invocationExpression.GetLocation(), EmptyMessageArgs));
-            }
-            else
+            if (invocation.TargetMethod.IsOverride)
             {
-                var lastIndex = arguments.Count - 1;
-                var lastArgumentTypeInfo = semanticModel.GetTypeInfo(arguments[lastIndex].Expression, cancellationToken);
-                if (lastArgumentTypeInfo.Type != null && !lastArgumentTypeInfo.Type.Equals(methodInfo.Parameters[lastIndex].Type))
+                var type = invocation.TargetMethod.ContainingType;
+                if (string.Equals(type.Name, "ValueType", StringComparison.OrdinalIgnoreCase) ||
+                    string.Equals(type.Name, "Enum", StringComparison.OrdinalIgnoreCase))
                 {
-                    reportDiagnostic(Diagnostic.Create(ParamsParameterRule, invocationExpression.GetLocation(), EmptyMessageArgs));
+                    context.ReportDiagnostic(Diagnostic.Create(ValueTypeNonOverridenCallRule, invocation.Syntax.GetLocation(), EmptyMessageArgs));
```

Comment:
You don't need the empty args, also please use the `CreateDiagnostic` helper method instead.

```suggestion
                    context.ReportDiagnostic(invocation.CreateDiagnostic(ValueTypeNonOverridenCallRule));
```